camp/camp-base/src/main/java/org/apache/brooklyn/camp/spi/collection/BasicResourceLookup.java

```diff
@@ -43,6 +47,11 @@ public synchronized void add(T item) {
         if (old!=null) {
             items.put(old.getId(), old);
             throw new IllegalStateException("Already contains item for "+item.getId()+": "+old+" (adding "+item+")");
```

Comment:
actually i get it, it puts the old value back before failing.  bit messy, have tidied.